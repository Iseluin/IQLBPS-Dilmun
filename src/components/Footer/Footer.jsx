import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { MdEmail } from 'react-icons/md';

import { HOME_ROUTE, ABOUT_ROUTE, FAQ_ROUTE } from '../../route';

const Footer = () => {
  const [email, setEmail] = useState(false);

  return (
    <footer className="p-4 bg-primary shadow md:px-6 md:py-6">
      <div className="sm:flex sm:items-center sm:justify-between">
        <Link to={HOME_ROUTE} className="flex items-center mb-4 sm:mb-0">
          <img
            src="https://cdn.discordapp.com/attachments/1031834305703460906/1035627738440159303/Asset_23.png"
            className="mr-3 h-10"
            alt="Flowbite Logo"
          />
          <span className="self-center text-2xl font-semibold text-background">
            Dilmun
          </span>
        </Link>
        <ul className="flex flex-wrap items-center mb-6 text-sm text-background sm:mb-0 font-semibold">
          <li>
            <Link
              to={HOME_ROUTE}
              className="mr-4 hover:underline hover:text-tertiary md:mr-6 "
            >
              Home
            </Link>
          </li>
          <li>
            <Link
              to={ABOUT_ROUTE}
              className="mr-4 hover:underline hover:text-tertiary md:mr-6"
            >
              Meet The Team
            </Link>
          </li>
          <li>
            <Link
              to={FAQ_ROUTE}
              className="mr-4 hover:underline hover:text-tertiary md:mr-6 "
            >
              FAQ
            </Link>
          </li>
          <li>
            <button type="button" onClick={() => setEmail(!email)}>
              {email ? (
                <p className="mr-4 hover:text-tertiary">
                  Dilmun.recoded@gmail.com
                </p>
              ) : (
                <MdEmail className="mr-4 hover:text-tertiary h-8 w-5" />
              )}
            </button>
          </li>
        </ul>
      </div>
      <hr className="my-6 border-secondary sm:mx-auto lg:my-8" />
      <span className="block text-sm text-tertiary sm:text-center">
        © 2022{' '}
        <Link to={HOME_ROUTE} className="hover:underline">
          Dilmun™
        </Link>
        . All Rights Reserved.
      </span>
    </footer>
  );
};

export default Footer;
