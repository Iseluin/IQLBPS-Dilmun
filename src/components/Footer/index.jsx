import React from 'react'

const Footer = () => {
    return (

        <div className="p-4 bg-white rounded-lg shadow md:px-6 md:py-8 dark:bg-tertiary">
            <div className="sm:flex sm:items-center sm:justify-between">

                <ul className="flex flex-wrap items-center mb-6 text-sm text-secondary sm:mb-0 dark:text-primary">
                    <li>
                        <a href="./" className="mr-4 hover:underline md:mr-6 ">About</a>
                    </li>
                    <li>
                        <a href="./" className="mr-4 hover:underline md:mr-6">Privacy Policy</a>
                    </li>
                    <li>
                        <a href="./" className="mr-4 hover:underline md:mr-6 ">Product Team</a>
                    </li>
                    <li>
                        <a href="./" className="hover:underline">Contact</a>
                    </li>
                </ul>
            </div>
            <hr className="my-6 border-gray-200 sm:mx-auto dark:border-gray-700 lg:my-8"

            />

            <span className="block text-sm text-primary sm:text-center dark:text-primary">© 2022 <a href="https://flowbite.com/" className="hover:underline">Flowbite™</a>. All Rights Reserved.
            </span>

        </div>



    )
}

export default Footer