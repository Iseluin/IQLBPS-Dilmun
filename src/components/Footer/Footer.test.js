import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import Footer from "./Footer";

test("redering footer", () => {
    const tree = renderer.create(<Router>
        <Footer />
    </Router>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("redering divs correctly correctly", () => {
    const { divs } = renderer.create(
        <Router>
            <div />
            <div />
            <div />
        </Router>).toJSON();
    expect(divs).toMatchSnapshot();
})

