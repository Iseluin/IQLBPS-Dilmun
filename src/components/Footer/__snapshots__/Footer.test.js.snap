// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redering divs correctly correctly 1`] = `undefined`;

exports[`redering footer 1`] = `
<footer
  className="p-4 bg-primary shadow md:px-6 md:py-6"
>
  <div
    className="sm:flex sm:items-center sm:justify-between"
  >
    <a
      className="flex items-center mb-4 sm:mb-0"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Flowbite Logo"
        className="mr-3 h-10"
        src="https://cdn.discordapp.com/attachments/1031834305703460906/1035627738440159303/Asset_23.png"
      />
      <span
        className="self-center text-2xl font-semibold text-background"
      >
        Dilmun
      </span>
    </a>
    <ul
      className="flex flex-wrap items-center mb-6 text-sm text-background sm:mb-0 font-semibold"
    >
      <li>
        <a
          className="mr-4 hover:underline hover:text-tertiary md:mr-6 "
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="mr-4 hover:underline hover:text-tertiary md:mr-6"
          href="/about"
          onClick={[Function]}
        >
          Meet The Team
        </a>
      </li>
      <li>
        <a
          className="mr-4 hover:underline hover:text-tertiary md:mr-6 "
          href="/faq"
          onClick={[Function]}
        >
          FAQ
        </a>
      </li>
      <li>
        <button
          onClick={[Function]}
          type="button"
        >
          <svg
            className="mr-4 hover:text-tertiary h-8 w-5"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </div>
  <hr
    className="my-6 border-secondary sm:mx-auto lg:my-8"
  />
  <span
    className="block text-sm text-tertiary sm:text-center"
  >
    © 2022
     
    <a
      className="hover:underline"
      href="/"
      onClick={[Function]}
    >
      Dilmun™
    </a>
    . All Rights Reserved.
  </span>
</footer>
`;
